### ISSUE TITLE
Deprecate @theia/git extension

### ISSUE THREAD

**Issue by JonasHelming** [eclipse-theia/theia#12745]
As discussed in several dev calls, we would like to deprecate the @theia/git extension. It has a functional overlap with the VS Code built-in Git extension and it does not make sense to maintain our own Git extension anymore. This will reduce maintenance efforts.
If you are an active user of the Theia Git extension, we recommend you to evaluate switching to the VS Code built-in. If you see any issues with the switch, please raise them here.

- [x] announce on mailing list
- [x] announce in news & noteworthy
- [x] announce in forum
- [x] blocker: unsupported menu paths: https://github.com/eclipse-theia/theia/issues/11730.
- [x] mark npm package as deprectated
- [x] remove code (packages/git)
- [x] add to migration guide

**Comment by vince-fugnitto** [comment]
Related https://github.com/eclipse-theia/theia/issues/7482.

**Comment by colin-grant-work** [comment]
I'd suggest that in addition to the issue that @vince-fugnitto has mentioned, any issues related to getting the VSCode builtin Git extensions & Gitlens working should be linked here so that adopters can be confident that the deprecation of `@theia/git` won't mean a loss of Git-related functionality.

**Comment by vince-fugnitto** [comment]
> any issues related to getting the VSCode builtin Git extensions & Gitlens working should be linked here

Such as missing menu support for the git builtins: https://github.com/eclipse-theia/theia/issues/11730.

**Comment by vince-fugnitto** [comment]
> Didn't we have a repo where "community-supported" packages would live after removing them from the main repo?

I'm not aware of any community-supported repo (although we can certainly create one if needed). I think the idea similarly to [theia-cpp-extensions](https://github.com/eclipse-theia/theia-cpp-extensions) in the past is to move the package to it's own repository where interested parties can contribute and maintain the package.

**Comment by Hanksha** [comment]
Hi, is it still worth reporting and fixing issues related to @theia/git ?

I was planning to use the vs code extension but it's not working yet and we're getting too close to our release to perform the switch. We found a few issues in @theia/git but since the package is being deprecated I'm not sure if I should contribute the fix or just implement a workaround in our project until we can use the vs code extension.

**Comment by JonasHelming** [comment]
@jfaltermeier It is not marked as deprecated anymore: https://www.npmjs.com/package/@theia/git
Do we have to repeat this for every new version maybe?

**Comment by jfaltermeier** [comment]
> @jfaltermeier It is not marked as deprecated anymore: https://www.npmjs.com/package/@theia/git Do we have to repeat this for every new version maybe?

The guide does not mention this case (https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions), but it looks like it. It will add this to the documentation

**Comment by jfaltermeier** [comment]
The package is no longer published to the npm registry.
We added a note about this in the breaking changes section of the changelog: https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md#breaking_changes_1.61.0
The source code remains available in the repository for now.

**Comment by struanb** [comment]
Is there a migration guide? I've tried to remove the plugin but struggled to arrange that our prebuilt Theia application (for the browser) opens with the vscode.git extension similarly preinstalled.

**Comment by msujew** [comment]
@struanb You can just add the [vscode.git](https://open-vsx.org/extension/vscode/git) and [vscode.git-base](https://open-vsx.org/extension/vscode/git-base) to the `theiaPlugins` directory of your package.json and run `theia download:plugins`.

We will likely add an entry to the migration guide soon.

**Comment by struanb** [comment]
Thank you @msujew. The way we package the Theia app, we launch it inside multiple different container environments. As such, the directory into which extensions should be placed before launch for our setup is (I suspect) _not_ the same directory that `theia download:plugins` writes to when the app is built. 

For avoidance of doubt, can you confirm which directory path will the built Theia app use to look for the vscode.git extension on launch, and is this directory path configurable?

**Comment by msujew** [comment]
@struanb The path needs to be configured via the `plugins` cli argument. See also:

https://github.com/eclipse-theia/theia/blob/19896b0638e740471583dafa6445b51e9793cb32/examples/browser/package.json#L102

**Comment by msujew** [comment]
@JonasHelming Do you think it makes sense (in light of https://github.com/eclipse-theia/theia/issues/16460) to work on the last two points of the issue?

**Comment by JonasHelming** [comment]
Yes, @ndoschek Could you schedule this in for one of the next milestones, please?
